pkg/extensions/lint/lint.go

```diff
@@ -53,17 +53,57 @@ func (linter *Linter) CheckMandatoryAnnotations(repo string, manifestDigest godi
 		return false, err
 	}
 
-	annotations := manifest.Annotations
+	mandatoryAnnotationsMap := make(map[string]bool)
+	for _, annotation := range mandatoryAnnotationsList {
+		mandatoryAnnotationsMap[annotation] = false
+	}
+
+	manifestAnnotations := manifest.Annotations
+	for annotation := range manifestAnnotations {
+		if _, ok := mandatoryAnnotationsMap[annotation]; ok {
+			mandatoryAnnotationsMap[annotation] = true
+		}
+	}
+
+	if checkAllAnnotationsFound(mandatoryAnnotationsMap) {
+		return true, nil
+	}
+
+	// if there are mandatory annotations missing in the manifest, get config and check these annotations too
+	configDigest := manifest.Config.Digest
+	content, err = imgStore.GetBlobContent(repo, string(configDigest))
+
+	if err != nil {
+		linter.log.Error().Err(err).Msg("linter: couldn't get config JSON" + string(configDigest))
+
+		return false, err
+	}
+
+	if err := json.Unmarshal(content, &manifest); err != nil {
+		linter.log.Error().Err(err).Msg("linter: couldn't unmarshal config JSON")
 
-	for _, annot := range mandatoryAnnotationsList {
-		_, found := annotations[annot]
+		return false, err
+	}
 
-		if !found {
-			// if annotations are not found, return false but it's not an error
-			linter.log.Error().Msgf("linter: missing %s annotations", annot)
+	configAnnotations := manifest.Config.Annotations
+	for annotation := range configAnnotations {
+		if _, ok := mandatoryAnnotationsMap[annotation]; ok {
+			mandatoryAnnotationsMap[annotation] = true
+		}
+	}
+
+	if !checkAllAnnotationsFound(mandatoryAnnotationsMap) {
```

Comment:
What if we have a single function getMissingAnnotations(annotations, mandatoryAnnotations map[string]bool)
```
		for annotation, flag := range mandatoryAnnotationsMap {
			if !flag {
				missingAnnotations = append(missingAnnotations, annotation)
			}
		}
```

and the check after manifest would be if len(missingAnnotations) == 0 { return true, nil}
check after config would be 
```
              if len(missingAnnotations) > 0 {
                   		linter.log.Error().Msgf("linter: missing %s annotations:", missingAnnotations)
                                return false, nil
               } 
```

Reply:
I implemented a new method as you suggested, replacing those checks in one method getMissingAnnotations()